e2e/yaml_test.go

```diff
@@ -0,0 +1,334 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+var (
+	ylog = logrus.WithField("component", "yaml_test")
+)
+
+func TestFlowFiltersYAML(t *testing.T) {
+	f1 := features.New("flow yaml").Setup(
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			output, err := RunCommand(ylog, "oc-netobserv", "flows",
+				"--protocol=TCP",
+				"--port=8080",
+				"or",
+				"--protocol=UDP",
+				"--yaml")
+			assert.Nil(t, err)
+
+			err = os.WriteFile(path.Join("output", StartupDate+"-flowYAMLOutput"), output, 0666)
+			assert.Nil(t, err)
+
+			str := string(output)
+			assert.NotEmpty(t, str)
+			// ensure script setup is fine
+			assert.Contains(t, str, "creating netobserv-cli namespace")
+			assert.Contains(t, str, "creating service account")
+			assert.Contains(t, str, "creating collector service")
+			assert.Contains(t, str, "creating flow-capture agents")
+			// check CLI done successfully
+			assert.Contains(t, str, "Check the generated YAML file in output folder")
+
+			return ctx
+		},
+	).Assess("check generated yaml output",
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			var yamls []string
+
+			dirPath := "output"
+			assert.True(t, dirExists(dirPath), "directory %s not found", dirPath)
+			err := filepath.Walk(dirPath, func(path string, info os.FileInfo, err error) error {
+				if err != nil {
+					fmt.Println(err)
+				}
+
+				if !info.IsDir() {
+					if strings.Contains(path, "flows_capture") && filepath.Ext(path) == ".yml" {
+						yamls = append(yamls, path)
+					}
+				}
+
+				return nil
+			})
+			assert.Nil(t, err)
+
+			// check yaml file
+			assert.Equal(t, 1, len(yamls))
+			yamlBytes, err := os.ReadFile(yamls[0])
+			assert.Nil(t, err)
+
+			// check yamls parts
+			yamlStr := string(yamlBytes[:])
+			yamls = strings.Split(yamlStr, "---")
+			assert.Equal(t, 6, len(yamls))
+
+			// check yaml contents
```

Comment:
this is hard to maintain why not create a multi line yaml content string and u can compare against it ?

Reply:
could you please provide an example ?

Reply:
here is an example of constant yaml string 
```golang
yaml := `
kind: Namespace
apiVersion: v1
metadata:
  name: netobserv-cli
  labels:
    app: netobserv
    pod-security.kubernetes.io/enforce: privileged
    pod-security.kubernetes.io/audit: privileged
    openshift.io/cluster-monitoring: "true"
`
```